viewing_party/party.py

```diff
@@ -1,23 +1,132 @@
+import statistics
+# -----------------------------------------
 # ------------- WAVE 1 --------------------
-
+# -----------------------------------------
 def create_movie(title, genre, rating):
-    pass
+    # create movie information
+    movie_info = {}
+    if title == None or genre == None or rating == None:
+        return None
+    movie_info['title'] = title
+    movie_info['genre'] = genre
+    movie_info['rating'] = rating
+    return movie_info 
+
+
+def add_to_watched(user_data, movie):
+    # add movie to what user has watched
+    user_data["watched"].append(movie)
+    return user_data
+
+
+def add_to_watchlist(user_data, movie):
+    # add movie to user's watchlist
+    user_data["watchlist"].append(movie)
+    return user_data
+
+
+def watch_movie(user_data, title):
+    # add a watched movie to watched
+    for movie in user_data["watchlist"]:
+        if movie["title"] == title:
+            user_data["watched"].append(movie)
+    # remove watched movie from watchlist
+    for movie in user_data["watchlist"]:
+        if movie["title"] == title:
+            user_data["watchlist"].remove(movie)
+    return user_data
 
 # -----------------------------------------
 # ------------- WAVE 2 --------------------
 # -----------------------------------------
+def get_watched_avg_rating(user_data):
+    combined_rating = 0
+    if user_data["watched"] == []:
+        average = 0.0
+    # calculate average of rating
+    for movie in user_data["watched"]:
+        combined_rating += movie["rating"]
+        average = (combined_rating/len(user_data["watched"]))
+    return average
+
 
+def get_most_watched_genre(user_data):
+    # make a list to store genre
+    mode_list = []
+    if len(user_data["watched"]) == 0:
+        return None
+    # use mode function to find genre mode of mode list
+    for movie in user_data["watched"]:
+        mode_list.append(movie["genre"])
+    return statistics.mode(mode_list)
 
 # -----------------------------------------
 # ------------- WAVE 3 --------------------
 # -----------------------------------------
+def get_unique_watched(user_data):
+    # make list of friends movies
+    friend_movies = []

```

Comment:
This was a good approach, to make your code even more efficient you could make `friend_movies` equal to a `set()` instead of an empty list. Then on `line 71` you would add it to the set instead of appending it to the list.